In [30]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sn
import numpy as np
import re

In [13]:
# load data food
df_food = pd.read_excel("food.xlsx")
df_food

(A1) CITY (A2) PROVINCE (B3) GENDER  (B4) AGE (B5) RELIGION (B6) ETHNIC  \
0     Jakarta   DKI Jakarta      Female        31         Islam        Java   
1     Jakarta   DKI Jakarta      Female        36         Islam        Java   
2     Jakarta   DKI Jakarta        Male        24         Islam        Java   
3     Jakarta   DKI Jakarta      Female        63         Islam       Other   
4     Jakarta   DKI Jakarta        Male        43    Protestant       Batak   
..        ...           ...         ...       ...           ...         ...   
705  Denpasar          Bali      Female        64         Islam        Java   
706  Denpasar          Bali        Male        51    Protestant        Java   
707  Denpasar          Bali        Male        26         Hindu       Batak   
708  Denpasar          Bali        Male        30         Hindu        Bali   
709  Denpasar          Bali      Female        45      Catholic        Bali   

           (B7) EDUCATION LEVEL  (B8) LIVED IN URBAN AREA (YEARS)  \
0            Senior High School                                31   
1               Bachelor degree                                32   
2            Senior High School                                24   
3            Senior High School                                32   
4                Primary School                                43   
..                          ...                               ...   
705          Junior High School                                24   
706  Master and doctoral degree                                24   
707          Senior High School                                26   
708          Senior High School                                30   
709          Senior High School                                23   

          (B9) OCCUPATION (B10) SOCIAL CLAS  ...  \
0       Private Employees       Lower Class  ...   
1       Private Employees       Lower Class  ...   
2       Private Employees      Middle Class  ...   
3       Private Employees       Lower Class  ...   
4    Government Employees       Lower Class  ...   
..                    ...               ...  ...   
705  Government Employees      Middle Class  ...   
706          Enterpreneur       Upper Class  ...   
707          Enterpreneur       Lower Class  ...   
708          Enterpreneur       Upper Class  ...   
709     Private Employees       Upper Class  ...   

     (E19) RELIGIOUS VALUE IN EATING PLACES  \
0                Neither agree nor disagree   
1                                     Agree   
2                                     Agree   
3                                     Agree   
4                            Strongly Agree   
..                                      ...   
705                          Strongly Agree   
706              Neither agree nor disagree   
707                                   Agree   
708              Neither agree nor disagree   
709                                   Agree   

    (E20) RELIGIOUS VALUE IN CONSUMPTION PATTERN  \
0                                          Agree   
1                                          Agree   
2                     Neither agree nor disagree   
3                     Neither agree nor disagree   
4                     Neither agree nor disagree   
..                                           ...   
705                   Neither agree nor disagree   
706                                        Agree   
707                               Strongly Agree   
708                               Strongly Agree   
709                                        Agree   

    (E21) CUSTOM VALUE IN TYPE OF FOOD (E22) CUSTOM VALUE IN EATING PLACES  \
0                       Strongly Agree          Neither agree nor disagree   
1                       Strongly Agree                               Agree   
2           Neither agree nor disagree                            Disagree   
3                                Agree          Neither agree nor disagree   
4           Neither a

In [6]:
#info dataset
df_food.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 710 entries, 0 to 709
Data columns (total 28 columns):
 #   Column                                                                 Non-Null Count  Dtype 
---  ------                                                                 --------------  ----- 
 0   (A1) CITY                                                              710 non-null    object
 1   (A2) PROVINCE                                                          710 non-null    object
 2   (B3) GENDER                                                            710 non-null    object
 3   (B4) AGE                                                               710 non-null    int64 
 4   (B5) RELIGION                                                          710 non-null    object
 5   (B6) ETHNIC                                                            710 non-null    object
 6   (B7) EDUCATION LEVEL                                                   710 non-null    object
 7  

## Cleansing Data

In [38]:
#change columns name
df_food.rename(columns=lambda x:re.sub(r'\([^)]*\)','',x).strip())

#change column
df_food.rename(columns={'(E27 TYPE OF DISHES VARIATION BASED ON RELIGIOUS RULES':'TYPE OF DISHES VARIATION BASED ON RELIGIOUS RULES'})

print(df_food.columns)

Index(['CITY', 'PROVINCE', 'GENDER', 'AGE', 'RELIGION', 'ETHNIC',
       'EDUCATION LEVEL', 'LIVED IN URBAN AREA', 'OCCUPATION', 'SOCIAL CLAS',
       'HOUSEHOLD MEMBER', 'HOUSEHOLD INCOME', 'TIME OF INCOME RECEIPT',
       'HOUSEHOLD EXPENDITURE', '% MONTHLY EXPENDITURE FOR FOOD',
       'HEALTH FACTOR', 'PRICE FACTOR)', 'RELIGIOUS VALUE IN TYPE OF FOOD',
       'RELIGIOUS VALUE IN EATING PLACES',
       'RELIGIOUS VALUE IN CONSUMPTION PATTERN',
       'CUSTOM VALUE IN TYPE OF FOOD', 'CUSTOM VALUE IN EATING PLACES',
       'CUSTOM VALUE IN CONSUMPTION PATTERN',
       'PRACTICE OF FOOD CONSUMPTION BASED ON RELIGIOUS RULES',
       'PRACTICE OF FOOD CONSUMPTION BASED ON CUSTOM RULES',
       'BUYING FOOD PRODUCT IN THE RIGHT PLACE BASED ON RELIGIOUS RULES',
       'TYPE OF DISHES VARIATION BASED ON RELIGIOUS RULES',
       'PRACTICE OF CONSUMING TRADITIONAL FOOD'],
      dtype='object')


In [16]:
#cek missing value food
#cek nilai null pada data train
#menghitung jumlah nilai NaN pada setiap kolom data frame df_train
null_valu = df_food.isna().sum().sort_values(ascending=False)

#menghitung persentase nilai NaN pada setiap kolom data frame df_train
pres_Nan = round(null_valu/len(df_food)*100,2)

#menyusun hasil perhitungan persentase nilai NaN pada setiap kolom ke dalam suatu data frame
#dengan kolom 'NaN %' dan menampilkan 5 baris pertama
pd.DataFrame({'NaN %':pres_Nan}).head()


NaN %
(A1) CITY                                             0.0
(A2) PROVINCE                                         0.0
(E27 TYPE OF DISHES VARIATION BASED ON RELIGIOU...    0.0
(E26) BUYING FOOD PRODUCT IN THE RIGHT PLACE BA...    0.0
(25) PRACTICE OF FOOD CONSUMPTION BASED ON CUST...    0.0

In [14]:
#cek outlier
print(f'Jumlah baris sebelum memfilter outlier: {len(df_food)}')

filtered_entries = np.array([True] * len(df_food))
for col in ['(B4) AGE','(B8) LIVED IN URBAN AREA (YEARS)','(B11) HOUSEHOLD MEMBER',
           '(C15) % MONTHLY EXPENDITURE FOR FOOD']:
    Q1 = df_food[col].quantile(0.25)
    Q3 = df_food[col].quantile(0.75)
    IQR = Q3 - Q1
    low_limit = Q1 - (IQR * 1.5)
    high_limit = Q3 + (IQR * 1.5)

    filtered_entries = ((df_food[col] >= low_limit) & (df_food[col] <= high_limit)) & filtered_entries
    
df_food = df_food[filtered_entries]

print(f'Jumlah baris setelah memfilter outlier: {len(df_food)}')

Jumlah baris sebelum memfilter outlier: 710
Jumlah baris setelah memfilter outlier: 419


In [23]:
#cek duplikasi data
df_food.duplicated(subset=['(A1) CITY','(A1) CITY','(A1) CITY','(A1) CITY','(B4) AGE','(B8) LIVED IN URBAN AREA (YEARS)','(B11) HOUSEHOLD MEMBER',
           '(C15) % MONTHLY EXPENDITURE FOR FOOD','(E23) CUSTOM VALUE IN CONSUMPTION PATTERN','(B11) HOUSEHOLD MEMBER',
                          '(B11) HOUSEHOLD MEMBER']).sum()

3

In [26]:
print(df_food.duplicated().sum())
df_food.drop_duplicates()
print(df_food.duplicated().sum())

0
0


In [27]:
#describe data numerik food
df_food.describe()

(B4) AGE  (B8) LIVED IN URBAN AREA (YEARS)  (B11) HOUSEHOLD MEMBER  \
count  419.000000                        419.000000                   419.0   
mean    40.961814                         27.976134                     6.0   
std     11.965400                          7.973905                     0.0   
min     22.000000                          7.000000                     6.0   
25%     31.000000                         23.000000                     6.0   
50%     38.000000                         28.000000                     6.0   
75%     52.000000                         33.000000                     6.0   
max     64.000000                         50.000000                     6.0   

       (C15) % MONTHLY EXPENDITURE FOR FOOD  
count                            419.000000  
mean                              41.947494  
std                               17.397055  
min                               18.000000  
25%                               26.000000  
50%                               38.000000  
75%                               58.000000  
max                               76.000000

In [29]:
#describe data object
df_food.describe(include='O')

(A1) CITY (A2) PROVINCE (B3) GENDER (B5) RELIGION (B6) ETHNIC  \
count        419           419         419           419         419   
unique         5             5           2             5           8   
top     Denpasar          Bali      Female         Islam        Java   
freq          99            99         246           254         177   

       (B7) EDUCATION LEVEL (B9) OCCUPATION (B10) SOCIAL CLAS  \
count                   419             419               419   
unique                    6               6                 3   
top                 Diploma    Enterpreneur       Lower Class   
freq                    153             120               145   

       (C12) HOUSEHOLD INCOME (MONTHLY IN MILLION IDR)  \
count                                              419   
unique                                              38   
top                                           03:04:00   
freq                                                32   

       (C13) TIME OF INCOME RECEIPT  ...  \
count                           419  ...   
unique                            3  ...   
top                           Daily  ...   
freq                            143  ...   

       (E19) RELIGIOUS VALUE IN EATING PLACES  \
count                                     419   
unique                                      4   
top                            Strongly Agree   
freq                                      165   

       (E20) RELIGIOUS VALUE IN CONSUMPTION PATTERN  \
count                                           419   
unique                                            4   
top                                  Strongly Agree   
freq                                            202   

       (E21) CUSTOM VALUE IN TYPE OF FOOD (E22) CUSTOM VALUE IN EATING PLACES  \
count                                 419                                 419   
unique                                  4                                   4   
top                                 Agree          Neither agree nor disagree   
freq                                  167                                 203   

       (E23) CUSTOM VALUE IN CONSUMPTION PATTERN  \
count                                        419   
unique                                         4   
top                   Neither agree nor disagree   
freq                                         199   

       (24) PRACTICE OF FOOD CONSUMPTION BASED ON RELIGIOUS RULES  \
count                                                 419           
unique                                                  4           
top                                                 Agree           
freq                                                  177           

       (25) PRACTICE OF FOOD CONSUMPTION BASED ON CUSTOM RULES  \
count                                                 419        
unique                                                  4        
top                            Neither agree nor disagree        
freq                                                  220        

       (E26) BUYING FOOD PRODUCT IN THE RIGHT PLACE BASED ON RELIGIOUS RULES  \
count                                                 419                      
unique                                                  4                      
top                            Neither agree nor disagree                      
freq                                                  184                      

       (E27 TYPE OF DISHES VARIATION BASED ON RELIGIOUS RULES  \
count                                                 419       
unique                                                  4       
top                            Neither agree nor disagree       
freq                                                  220       

       (E28) PRACTICE OF CONSUMING TRADITIONAL FOOD  
count                                           419  
unique                                            4  
top                      Neither agree nor disagree  

### Analisa data

Dari data yang sudah dideskripsikan, berikut beberapa informasi dasar yang dapat diambil:
* umur orang yang ada pada survey rata-rata berumur 40 tahun dengan rata-rata lama tinggal di daerah urban selama 27 tahun dan presentase biaya bulanan yang dihabiskan untuk makanan bulanan dalam persen yaitu rata-rata 41%
* daerah pengisi survey paling banyak dari kota denpasar, provinsi bali
* gender pengisi survey didominasi oleh perempuan dengan total 246 pengisi dari 419
* jenjang pengisi survey paling banyak dari tingkat diploma

berikut visualisasi variabel yang ada untuk dilihat keterkaitan antar data: